exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -281,97 +401,81 @@ func (e *Exchange) UpdateOrderbook(ctx context.Context, pair currency.Pair, asse
-			}
-		}
-		err = book.Process()
+		book.Bids = orderbookLevelFromSlice(orderbookNew.Bids)
+		book.Asks = orderbookLevelFromSlice(orderbookNew.Asks)
+	case asset.Futures:
+		var orderbookNew *FuturesOrderbook
+		orderbookNew, err = e.GetFuturesOrderBook(ctx, fPair.String(), 0, 150)
 		if err != nil {
-			return book, err
+			return nil, err
 		}
+		book.Bids = orderbookNew.Bids.Levels()
+		book.Asks = orderbookNew.Asks.Levels()
+	default:
+		return nil, fmt.Errorf("%w asset type: %v", asset.ErrNotSupported, assetType)
+	}
+	if err := book.Process(); err != nil {
+		return book, err
 	}
-	return orderbook.Get(e.Name, pair, assetType)
+	return orderbook.Get(e.Name, fPair, assetType)
 }
 
 // UpdateAccountInfo retrieves balances for all enabled currencies for the
 // Poloniex exchange
 func (e *Exchange) UpdateAccountInfo(ctx context.Context, assetType asset.Item) (account.Holdings, error) {
 	var response account.Holdings
-	response.Exchange = e.Name
-	accountBalance, err := e.GetBalances(ctx)
-	if err != nil {
-		return response, err
-	}
-
-	currencies := make([]account.Balance, 0, len(accountBalance.Currency))
-	for x, y := range accountBalance.Currency {
-		currencies = append(currencies, account.Balance{
-			Currency: currency.NewCode(x),
-			Total:    y,
-		})
-	}
-
-	response.Accounts = append(response.Accounts, account.SubAccount{
-		AssetType:  assetType,
-		Currencies: currencies,
-	})
-
-	creds, err := e.GetCredentials(ctx)
-	if err != nil {
-		return account.Holdings{}, err
-	}
-	err = account.Process(&response, creds)
-	if err != nil {
-		return account.Holdings{}, err
+	switch assetType {
+	case asset.Spot, asset.Futures:
+		accountBalance, err := e.GetSubAccountBalances(ctx)
+		if err != nil {
+			return response, err
+		}
+		subAccounts := make([]account.SubAccount, len(accountBalance))
+		for i := range accountBalance {
+			subAccount := account.SubAccount{
+				ID:        accountBalance[i].AccountID,
+				AssetType: stringToAccountType(accountBalance[i].AccountType),
+			}
+			currencyBalances := make([]account.Balance, len(accountBalance[i].Balances))
+			for x := range accountBalance[i].Balances {
+				currencyBalances[x] = account.Balance{
+					Currency:               currency.NewCode(accountBalance[i].Balances[x].Currency),
+					Total:                  accountBalance[i].Balances[x].AvailableBalance.Float64(),
+					Hold:                   accountBalance[i].Balances[x].Hold.Float64(),
+					Free:                   accountBalance[i].Balances[x].Available.Float64(),
+					AvailableWithoutBorrow: accountBalance[i].Balances[x].AvailableBalance.Float64(),
+				}
+			}
+			subAccounts[i] = subAccount
+		}
+		response = account.Holdings{
+			Exchange: e.Name,
+			Accounts: subAccounts,
+		}
+		creds, err := e.GetCredentials(ctx)
+		if err != nil {
+			return response, err
+		}
+		return response, account.Process(&response, creds)
+	default:
+		return response, fmt.Errorf("%w: asset type: %q", asset.ErrNotSupported, assetType)
```

Comment:
🚧 Change request:
Sorry, didn't spot this last time, but:
```suggestion
		return response, fmt.Errorf("%w: %q", asset.ErrNotSupported, assetType)
```